DEVELOPMENT.md

```diff
@@ -38,4 +38,20 @@ for gapic-generator-java's Bazel build.
 
     ```sh
     mvn fmt:format
-    ```
\ No newline at end of file
+    ```
+
+## Testing guide
+There are 4 layers of testing in this repo:
+1. Traditional unit tests
+2. Golden unit tests
+3. Showcase integration tests
```

Comment:
Could be helpful to have a quick blurb specify that:
Golden unit tests -> Generate an output file for the specific composer 
Showcase integration -> Test the generated library's behavior against a mock server (and then link to the showcase module's README/ DEVELOPMENT.md)

Especially helpful for a first-time contributor

Reply:
Thanks, will do! I was planning to add more detailed explanation but wanted to get some initial feedback first, glad to see someone else also found it useful.

Reply:
Updated, PTAL.